### PR TITLE
CROS-174: Add Action Test

### PR DESCRIPTION
**Pull request by RuilaiZhang**

### CODE REVIEW

File: installDep.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+cd node_modules/delphinus-curves
+npm install
```
Comment (yymone) [inline installDep.sh:2]: need change to npm run prepare, or it will duplicate install node_module locally.

File: installDep.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+cd -
+
+cd node_modules/web3subscriber
+npm install
```
Comment (yymone) [inline installDep.sh:6]: same

File: tests/depositWithdrawTest.ts [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        data = data.toHex();
+      }
+      return new BN(data, 16);
+    }
+    let nonce = 0;
+    let accountIndex = 0;
+    let tokenIndex = tokenIdx;
+    let amount = 1;
+    let config = await getConfigByChainName(L1ClientRole.Monitor, testChain);
+    let l1address = encodeL1address("4D9A852e6AECD3A6E87FecE2cA109780E45E6F2D", parseInt(config.deviceId).toString(16));
```
Comment (yymone) [inline tests/depositWithdrawTest.ts:185]: why we have hard code l1 address here?
I can see in the mintToken function you use config.monitorAccount. So here should not be hardcode?

File: tests/depositWithdrawTest.ts [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+      });
+    });
+    await r.when(
+      "mint",
+      "transactionHash",
+      (hash: string) => console.log(hash)
+    );
+  }
+
+async function addToken(testChain: string){
```
Comment (yymone) [inline tests/depositWithdrawTest.ts:44]: This one is not good.
For testing purpose PROXY contract, we can do this.
However for production, we cannot "adding a token to the production PROXY contract when we run test"
Thus better find a way to do "try to find the index of the TOKEN in the PROXY contract" and then use th index.
I think we probably can do this as we can get the details of tokens from the proxy (that status.ts had done that) and then compare the address with TOKEN address and find the index. (If multiple, just use the first one?)

Comment (yymone) [reply tests/depositWithdrawTest.ts:44]: Ummm, anyway this is not that important as we have to use test purpose PROXY contract to do the testing as we send mock data to the PROXY.

So this scenario testing should to be done on testing PROXY contract only.

So we do not need change this one.

However, I think we need remind tests runner about this.
So I think we need write a reminder in the actionTest.sh and ask the user that PROXY contract address in PROXY.json is test purpose only and not production address, and verifier is Dummy. Until user press "Y", then we begin the testing.

Comment (RuilaiZhang) [reply tests/depositWithdrawTest.ts:44]: this function will check and add test token only if there isn't one, no other tokens.

**Pull request merged**